generate-images.sh

```diff
@@ -7,32 +7,35 @@ OUTPUT=output
 WATERMARK='© 2nd Story Toolkit'
 
 stamp() {
-  # generate watermark 
+  # generate watermark
+  # see https://amytabb.com/til/photography/2021/01/23/image-magick-watermark/ for reference
   mkdir -p stamp
   pushd stamp || return
-  convert -size 600x100 xc:grey30 -font Arial -pointsize 40 -gravity center \
-    -draw "fill grey70  text 0,0  '$WATERMARK'" \
+  convert -size 1800x300 xc:transparent -font "Roboto-Bold-Italic" -pointsize 100 -gravity center \
+      -fill black        -annotate +24+64 "$WATERMARK" \
+      -fill white        -annotate +26+66 "$WATERMARK" \
+      -fill transparent  -annotate +25+65 "$WATERMARK" \
     stamp_fgnd.png
-  convert -size 600x100 xc:black -font Arial -pointsize 40 -gravity center \
-    -draw "fill white  text  1,1  '$WATERMARK'  \
-                                text  0,0  '$WATERMARK'  \
-                    fill black  text -1,-1 '$WATERMARK'" \
-    +matte stamp_mask.png
-  composite -compose CopyOpacity stamp_mask.png stamp_fgnd.png stamp.png
+  convert -size 1800x300 xc:black -font "Roboto-Bold-Italic" -pointsize 100 -gravity center \
+      -fill white   -annotate +24+64 "$WATERMARK" \
+      -fill white   -annotate +26+66 "$WATERMARK" \
+      -fill black   -annotate +25+65 "$WATERMARK" \
+    stamp_mask.jpg
+  composite -compose CopyOpacity stamp_mask.jpg stamp_fgnd.png stamp.png
   mogrify -trim +repage stamp.png
   popd || return
 }
 
 watermark() {
   filename=$1
-  composite -gravity southeast -geometry +0+10 stamp/stamp.png  "$filename" "$filename"
+  composite -gravity southeast -geometry +40+30 stamp/stamp.png  "$filename" "$filename"
 }
 
 number() {
   filename=$1
   number=$2
   output="$OUTPUT"/"$number".jpeg
-  convert "$filename" -gravity SouthWest -pointsize 250 -fill black -annotate 0 "$number" "$output"
+  convert "$filename" -resize 3400x2550 -gravity SouthWest -pointsize 100 -fill black -font "Roboto-Bold" -annotate +40+30 "$number" "$output"
```

Comment:
I adjusted the size of the image here but it's pretty big for your use case. Curious what the lowest pixel size is that you can get away with. Maybe something like 1333x1000?

And the `+40+30` is adjusting the bottom and right-hand margins from the watermark.